plutus-ledger-api/src/PlutusLedgerApi/Data/V1.hs

```diff
@@ -0,0 +1,219 @@
+  TxOutRef (..),
+  TxInInfo (..),
+
+  -- *** Intervals
+  Interval (..),
+  Extended (..),
+  Closure,
+  UpperBound (..),
+  LowerBound (..),
+  always,
+  from,
+  to,
+  lowerBound,
+  upperBound,
+  strictLowerBound,
+  strictUpperBound,
+
+  -- *** Newtypes and hash types
+  ScriptHash (..),
+  Redeemer (..),
+  RedeemerHash (..),
+  Datum (..),
+  DatumHash (..),
+
+  -- * Data
+  PLC.Data (..),
+  BuiltinData (..),
+  ToData (..),
+  FromData (..),
+  UnsafeFromData (..),
+  toData,
+  fromData,
+  dataToBuiltinData,
+  builtinDataToData,
+
+  -- * Errors
+  EvaluationError (..),
+  ScriptDecodeError (..),
+) where
+
+import Data.SatInt
+import PlutusCore.Data qualified as PLC
+import PlutusCore.Evaluation.Machine.ExBudget as PLC
+import PlutusLedgerApi.Common as Common hiding (deserialiseScript, evaluateScriptCounting,
+                                         evaluateScriptRestricting)
+import PlutusLedgerApi.Common qualified as Common (deserialiseScript, evaluateScriptCounting,
+                                                   evaluateScriptRestricting)
+import PlutusLedgerApi.V1.Address
+import PlutusLedgerApi.V1.Bytes
+import PlutusLedgerApi.V1.Credential
+import PlutusLedgerApi.V1.Crypto
+import PlutusLedgerApi.V1.Data.Contexts
+import PlutusLedgerApi.V1.Data.Value
+import PlutusLedgerApi.V1.DCert
+import PlutusLedgerApi.V1.EvaluationContext
+import PlutusLedgerApi.V1.Interval hiding (singleton)
+import PlutusLedgerApi.V1.ParamName
+import PlutusLedgerApi.V1.Scripts as Scripts
+import PlutusLedgerApi.V1.Time
+
+{- | An alias to the Plutus ledger language this module exposes at runtime.
+ MAYBE: Use CPP '__FILE__' + some TH to automate this.
+-}
+thisLedgerLanguage :: PlutusLedgerLanguage
+thisLedgerLanguage = PlutusV1
+
+{- Note [Abstract types in the ledger API]
+We need to support old versions of the ledger API as we update the code that it depends on. You
+might think that we should therefore make the types that we expose abstract, and only expose
+specific functions for constructing and working with them. However the situation is slightly
+different for us.
+
+Normally, when you are in this situation, you want to retain the same *interface* as the old version,
+but with the new types and functions underneath. Abstraction lets you do this easily. But we actually
+want to keep the old *implementation*, because things really have to work the same, bug-for-bug. And
+the types have to translate into Plutus Core in exactly the same way, and so on.
+
+So we're going to end up with multiple versions of the types and functions that we expose here, even
+internally. That means we don't lose anything by exposing all the details: we're never going to remove
+anything, we're just going to create new versions.
+-}
+
+{- | The deserialization from a serialised script into a `ScriptForEvaluation`,
+ready to be evaluated on-chain.
+Called inside phase-1 validation (i.e., deserialisation error is a phase-1 error).
+-}
+deserialiseScript ::
```

Comment:
I don't see why we need to redefine this. Can we not re-export the existing one?

Reply:
Duplicating a function that uses map would be one thing, but duplicating something that doesn't use map at all makes little sense. Consider making `PlutusLedgerApi.V1` agnostic of the choice of map/list, and making `PlutusLedgerApi.V1.Term` and `PlutusLedgerApi.V1.Data` for anything specific to a particular map/list.